src/blockstore/rocksdb.zig

```diff
@@ -0,0 +1,193 @@
+
+            // initialize handle slice
+            for (1..cfs.len) |i| {
+                cf_handles[i - 1] = cfs[i].handle;
+            }
+
+            return .{
+                .allocator = allocator,
+                .db = database,
+                .logger = logger,
+                .cf_handles = cf_handles,
+            };
+        }
+
+        pub fn deinit(self: *Self) void {
+            self.allocator.free(self.cf_handles);
+            self.db.deinit();
+        }
+
+        pub fn put(
+            self: *Self,
+            comptime cf: ColumnFamily,
+            key: cf.Key,
+            value: cf.Value,
+        ) Error!void {
+            const key_bytes = try serializeToRef(self.allocator, key);
+            defer key_bytes.deinit();
+            const val_bytes = try serializeToRef(self.allocator, value);
+            defer val_bytes.deinit();
+            return try callRocks(
+                self.logger,
+                rocks.DB.put,
+                .{
+                    &self.db,
+                    self.cf_handles[cf.find(column_families)],
+                    key_bytes.data,
+                    val_bytes.data,
+                },
+            );
+        }
+
+        pub fn get(self: *Self, comptime cf: ColumnFamily, key: cf.Key) Error!?cf.Value {
+            const val_bytes = try self.getBytes(cf, key) orelse return null;
+            defer val_bytes.deinit();
+            return try deserialize(cf.Value, self.allocator, val_bytes.data);
+        }
+
+        pub fn getBytes(self: *Self, comptime cf: ColumnFamily, key: cf.Key) Error!?BytesRef {
+            const key_bytes = try serializeToRef(self.allocator, key);
+            defer key_bytes.deinit();
+            const val_bytes: rocks.Data = try callRocks(
+                self.logger,
+                rocks.DB.get,
+                .{ &self.db, self.cf_handles[cf.find(column_families)], key_bytes.data },
+            ) orelse return null;
+            return .{
+                .allocator = val_bytes.allocator,
+                .data = val_bytes.data,
+            };
+        }
+
+        pub fn delete(self: *Self, comptime cf: ColumnFamily, key: cf.Key) Error!void {
+            const key_bytes = try serializeToRef(self.allocator, key);
+            defer key_bytes.deinit();
+            return try callRocks(
+                self.logger,
+                rocks.DB.delete,
+                .{ &self.db, self.cf_handles[cf.find(column_families)], key_bytes.data },
+            );
+        }
+
+        pub fn writeBatch(self: *Self) Error!WriteBatch {
+            return .{
+                .inner = rocks.WriteBatch.init(),
+                .cf_handles = self.cf_handles,
+            };
+        }
+
+        pub fn commit(self: *Self, batch: WriteBatch) Error!void {
+            return callRocks(self.logger, self.db.write, .{batch.inner});
+        }
+
+        pub const WriteBatch = struct {
```

Comment:
can you add a comment explaining more context on why we have this WriteBatch interface -- the code seems to be the exact same as the outer struct (`WriteBatch.put` and `RocksDB.put`)

Reply:
8b93fa92ed066a8cb70bc80745f33a2951cbfc3c